src/signalhandler.cc

```diff
@@ -53,6 +53,8 @@
 #  include <sys/ucontext.h>
 #endif
 #ifdef HAVE_UNISTD_H
+#  include <sys/syscall.h>
```

Comment:
The availability of a POSIX header (`unistd.h`) does not imply the availability of LInux headers. Please use the appropriate config macros (i.e., `HAVE_SYS_SYSCALL_H`, etc.) here.